pkg/util/containers/v2/metrics/collector_docker.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+//go:build docker && (linux || windows)
+// +build docker,linux docker,windows
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/docker/docker/api/types"
+)
+
+const (
+	dockerCollectorID = "docker"
+)
+
+func init() {
+	metricsProvider.registerCollector(collectorMetadata{
+		id:       dockerCollectorID,
+		priority: 1,
+		runtimes: []string{"RuntimeNameDocker"},
+		factory: func() (Collector, error) {
+			return newDockerCollector()
+		},
+	})
+}
+
+type dockerCollector struct {
+	du *docker.DockerUtil
+}
+
+func newDockerCollector() (*dockerCollector, error) {
+	du, err := docker.GetDockerUtil()
```

Comment:
Thinking that it should check first if the Docker feature is activated or not

Reply:
Check added !